base/iostream.jl

```diff
@@ -229,6 +229,17 @@ function readline(s::IOStream)
     ccall(:jl_readuntil, Ref{String}, (Ptr{Void}, UInt8, UInt8), s.ios, '\n', 1)
 end
 
+"""
+    readuntil_string(s::IO, delim::UInt8)
+
+Like `readuntil(s, delim)`, but returns a `String` rather than
+a `Vector{UInt8}`.
+"""
+readuntil_string(s::IO, delim::UInt8) = String(readuntil(s, delim))
+function readuntil_string(s::IOStream, delim::UInt8)
```

Comment:
if `readuntil(io, delim::UInt8)` is changed to return a string, people could just use `Vector{UInt8}(readuntil(io, delim))` to get a `UInt8` array (and it will still work in Julia 0.5), so maybe we won't need to define a separate method.